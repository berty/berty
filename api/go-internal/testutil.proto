syntax = "proto3";

package testutil;

option go_package = "berty.tech/berty/go/internal/testutil";

service TestService {
  // use to test unary
  rpc EchoTest(EchoTest.Request) returns (EchoTest.Reply);

  // Use to test stream.
  rpc EchoStreamTest(EchoStreamTest.Request) returns (stream EchoStreamTest.Reply);

  // Use to test duplex stream.
  rpc EchoDuplexTest(stream EchoDuplexTest.Request) returns (stream EchoDuplexTest.Reply);
}

message EchoTest {
  message Request {
    int64 delay = 1;
    string echo = 2;
    bool trigger_error = 3;
  }
  message Reply {
    string echo = 1;
  }
}

message EchoStreamTest {
  message Request {
    int64 delay = 1;
    string echo = 2;
    bool trigger_error = 3;
  }
  message Reply {
    string echo = 1;
  }
}

message EchoDuplexTest {
  message Request {
    string echo = 1;
    bool trigger_error = 2;
  }
  message Reply {
    string echo = 1;
  }
}
