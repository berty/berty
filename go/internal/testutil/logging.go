package testutil

import (
	"flag"
	"os"
	"strings"
	"sync"
	"testing"

	"github.com/stretchr/testify/assert"
	"go.uber.org/zap"

	"berty.tech/berty/v2/go/internal/logutil"
)

const defaultLoggingFilters = "info+:bty.test* error+:*,-ipfs*"

var (
	logFilters     = flag.String("log-filters", defaultLoggingFilters, "log namespaces")
	logFile        = flag.String("log-file", "", "log to file")
	logFormat      = flag.String("log-format", "color", "json, console, color")
	loggerInstance *zap.Logger
	loggerCleanup  func()
	loggerInitOnce sync.Once
)

func Logger(t *testing.T) (*zap.Logger, func()) {
	t.Helper()

	loggerInitOnce.Do(func() {
		if val := os.Getenv("BERTY_LOGFILTERS"); val != "" {
			*logFilters = val
		}
		if val := os.Getenv("BERTY_LOGFILE"); val != "" {
			*logFile = val
		}
		if val := os.Getenv("BERTY_LOGFORMAT"); val != "" {
			*logFormat = val
		}
		*logFilters = strings.ReplaceAll(*logFilters, ":default:", defaultLoggingFilters)

		var err error
		loggerInstance, loggerCleanup, err = logutil.NewLogger(*logFilters, *logFormat, *logFile)
		if !assert.NoError(t, err) {
			loggerInstance = zap.NewNop()
			loggerCleanup = func() {}
		}
	})

	return loggerInstance, loggerCleanup
}
