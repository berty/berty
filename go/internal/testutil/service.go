package testutil

import (
	context "context"
	"testing"
	"time"

	"github.com/stretchr/testify/assert"
	"go.uber.org/zap"
	"google.golang.org/grpc"

	"berty.tech/berty/v2/go/pkg/errcode"
	"berty.tech/weshnet/v2/pkg/grpcutil"
)

type service struct {
	logger *zap.Logger

	UnimplementedTestServiceServer
}

type Options struct {
	Logger *zap.Logger
}

func (o *Options) applyDefault() {
	if o.Logger == nil {
		o.Logger = zap.NewNop()
	}
}

func TestingNewServiceClient(ctx context.Context, t *testing.T, opts *Options) (*grpc.ClientConn, *grpc.Server) {
	t.Helper()

	srv := grpc.NewServer()
	svc := NewService(opts)
	RegisterTestServiceServer(srv, svc)

	l := grpcutil.NewBufListener(2048)

	cc, err := l.NewClientConn(ctx)
	assert.NoError(t, err)

	go func() {
		_ = srv.Serve(l.Listener)
	}()

	t.Cleanup(func() { l.Close() })

	return cc, srv
}

func NewService(opts *Options) TestServiceServer {
	opts.applyDefault()
	return &service{
		logger: opts.Logger,
	}
}

func (svc *service) EchoTest(_ context.Context, req *EchoTest_Request) (*EchoTest_Reply, error) {
	if req.TriggerError {
		return nil, errcode.ErrCode_ErrTestEcho
	}

	if req.Delay > 0 {
		time.Sleep(time.Duration(req.Delay) * time.Millisecond)
	}

	return &EchoTest_Reply{Echo: req.Echo}, nil
}

func (svc *service) EchoStreamTest(req *EchoStreamTest_Request, srv TestService_EchoStreamTestServer) error {
	if req.TriggerError {
		return errcode.ErrCode_ErrTestEcho
	}

	for {
		err := srv.Send(&EchoStreamTest_Reply{Echo: req.Echo})
		if err != nil {
			return errcode.ErrCode_ErrTestEchoSend.Wrap(err)
		}

		time.Sleep(time.Duration(req.Delay) * time.Millisecond)
	}
}

func (svc *service) EchoDuplexTest(srv TestService_EchoDuplexTestServer) error {
	for {
		req, err := srv.Recv()
		if err != nil {
			return errcode.ErrCode_ErrTestEchoRecv.Wrap(err)
		}

		if req.TriggerError {
			return errcode.ErrCode_ErrTestEcho
		}

		err = srv.Send(&EchoDuplexTest_Reply{
			Echo: req.Echo,
		})
		if err != nil {
			return errcode.ErrCode_ErrTestEchoSend.Wrap(err)
		}
	}
}
