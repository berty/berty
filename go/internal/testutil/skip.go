package testutil

import (
	"os"
	"testing"
)

func SkipSlow(t *testing.T) {
	t.Helper()
	if os.Getenv("SKIP_SLOW_TESTS") == "1" {
		t.Skip("slow test skipped")
	}
}
