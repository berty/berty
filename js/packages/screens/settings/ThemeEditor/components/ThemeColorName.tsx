import { BlurView } from '@react-native-community/blur'
import React from 'react'
import { useTranslation } from 'react-i18next'
import { View, TouchableOpacity, StyleSheet } from 'react-native'
import { useDispatch } from 'react-redux'

import Avatar from '@berty/assets/logo/buck_berty_icon_card.svg'
import {
	TwoHorizontalButtons,
	PrimaryButtonIconLeft,
	SecondaryButtonIconLeft,
} from '@berty/components'
import { SmallInput } from '@berty/components'
import { UnifiedText } from '@berty/components/shared-components/UnifiedText'
import { useAppDimensions } from '@berty/contexts/app-dimensions.context'
import { useStyles } from '@berty/contexts/styles'
import { saveTheme } from '@berty/redux/reducers/theme.reducer'
import { useThemeColor } from '@berty/store'

const ThemeColorBody: React.FC<{ closeModal: () => void }> = ({ closeModal }) => {
	const [themeName, setThemeName] = React.useState<string>('')
	const { text, margin, padding, border } = useStyles()
	const { scaleHeight, scaleSize } = useAppDimensions()
	const { row, text, margin, padding, border, opacity } = useStyles()
	const { scaleHeight } = useAppDimensions()
	const colors = useThemeColor()
	const { t } = useTranslation()
	const dispatch = useDispatch()

	return (
		<View
			style={[
				{
					justifyContent: 'center',
					alignItems: 'center',
					height: 250 * scaleHeight,
					top: '25%',
				},
				margin.big,
			]}
		>
			<View
				style={[
					{
						width: 110 * scaleHeight,
						height: 110 * scaleHeight,
						backgroundColor: colors['main-background'],
						justifyContent: 'center',
						alignItems: 'center',
						position: 'relative',
						top: 50 * scaleHeight,
						zIndex: 1,
						elevation: 7,
						shadowOpacity: 0.1,
						shadowRadius: 5,
						shadowColor: colors.shadow,
						shadowOffset: { width: 0, height: 3 },
					},
					border.radius.scale(60),
				]}
			>
				<Avatar width={125 * scaleHeight} height={125 * scaleHeight} />
			</View>
			<View
				style={[
					padding.horizontal.medium,
					padding.bottom.medium,
					border.radius.large,
					border.shadow.huge,
					{ backgroundColor: colors['main-background'], shadowColor: colors.shadow },
				]}
			>
				<View style={[margin.top.scale(70 * scaleHeight)]}>
					<UnifiedText style={[text.align.center, padding.top.small, text.size.large, text.bold]}>
						{`ðŸŽ¨ ${t('modals.save-theme.title')}`}
					</UnifiedText>
					<View
						style={[padding.top.scale(20), padding.horizontal.medium, { flexDirection: 'column' }]}
					>
						<View>
							<UnifiedText style={[text.light]}>{t('modals.save-theme.desc')}</UnifiedText>
						</View>

						<View style={[margin.top.medium]}>
							<SmallInput
								value={themeName}
								onChangeText={setThemeName}
								placeholder={t('modals.save-theme.placeholder')}
							/>
						</View>
					</View>
				</View>

				<View style={[margin.top.medium, margin.horizontal.medium]}>
					<TwoHorizontalButtons>
						<SecondaryButtonIconLeft name='close' onPress={closeModal}>
							{t('modals.save-theme.cancel')}
						</SecondaryButtonIconLeft>
						<PrimaryButtonIconLeft
							onPress={() => {
								dispatch(saveTheme({ themeName }))
								closeModal()
							}}
						>
							{t('modals.save-theme.add')}
						</PrimaryButtonIconLeft>
					</TwoHorizontalButtons>
				</View>
			</View>
		</View>
	)
}

export const ThemeColorName: React.FC<{ closeModal: () => void }> = ({ closeModal }) => {
	const { windowHeight } = useAppDimensions()

	return (
		<View style={[StyleSheet.absoluteFill, { elevation: 6, zIndex: 6 }]}>
			<TouchableOpacity
				activeOpacity={0}
				style={[
					StyleSheet.absoluteFill,
					{
						position: 'absolute',
						top: 0,
						bottom: 0,
						left: 0,
						right: 0,
						height: windowHeight,
					},
				]}
				onPress={async () => {
					closeModal()
				}}
			>
				<BlurView style={[StyleSheet.absoluteFill]} blurType='light' />
			</TouchableOpacity>
			<ThemeColorBody closeModal={closeModal} />
		</View>
	)
}
