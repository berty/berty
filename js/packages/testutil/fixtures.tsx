import beapi from '@berty-tech/api'

const empty: beapi.messenger.IMedia = {
	metadataBytes: beapi.messenger.MediaMetadata.encode({}).finish(),
}

const media = {
	empty,
}

export default {
	media,
}
