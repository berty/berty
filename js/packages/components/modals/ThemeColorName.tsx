import React from 'react'
import { View, TouchableOpacity, Text as TextNative, StyleSheet, TextInput } from 'react-native'
import { Icon } from '@ui-kitten/components'
import { BlurView } from '@react-native-community/blur'
import { useTranslation } from 'react-i18next'

import { useStyles } from '@berty-tech/styles'
import {
	useMessengerContext,
	useThemeColor,
	CurrentGeneratedTheme,
	PersistentOptionsKeys,
	DefaultDarkTheme,
} from '@berty-tech/store'

import Avatar from './Buck_Berty_Icon_Card.svg'
import { useStylesDefaultModal } from './AddBot'

export const ThemeColorBody: React.FC<{ closeModal: () => void }> = ({ closeModal }) => {
	const [themeName, setThemeName] = React.useState<string>('')
	const [{ row, text, margin, padding, border, opacity }, { scaleHeight, scaleSize }] = useStyles()
	const colors = useThemeColor()
	const _styles = useStylesDefaultModal()
	const ctx = useMessengerContext()
	const { t } = useTranslation()

	return (
		<View
			style={[
				{
					justifyContent: 'center',
					alignItems: 'center',
					height: 250 * scaleHeight,
					top: '25%',
				},
				margin.big,
			]}
		>
			<View
				style={[
					{
						width: 110 * scaleHeight,
						height: 110 * scaleHeight,
						backgroundColor: colors['main-background'],
						justifyContent: 'center',
						alignItems: 'center',
						position: 'relative',
						top: 50 * scaleHeight,
						zIndex: 1,
						elevation: 7,
						shadowOpacity: 0.1,
						shadowRadius: 5,
						shadowColor: colors.shadow,
						shadowOffset: { width: 0, height: 3 },
					},
					border.radius.scale(60),
				]}
			>
				<Avatar width={125 * scaleHeight} height={125 * scaleHeight} />
			</View>
			<View
				style={[
					padding.horizontal.medium,
					padding.bottom.medium,
					border.radius.large,
					border.shadow.huge,
					{ backgroundColor: colors['main-background'], shadowColor: colors.shadow },
				]}
			>
				<View style={[margin.top.scale(70 * scaleHeight)]}>
					<TextNative
						style={[
							text.align.center,
							padding.top.small,
							text.size.large,
							text.bold.medium,
							{ fontFamily: 'Open Sans', color: colors['main-text'] },
						]}
					>
						{`ðŸŽ¨ ${t('modals.save-theme.title')}`}
					</TextNative>
					<View
						style={[padding.top.scale(20), padding.horizontal.medium, { flexDirection: 'column' }]}
					>
						<View>
							<TextNative
								style={[
									text.bold.small,
									text.size.medium,
									{ fontFamily: 'Open Sans', color: colors['main-text'] },
								]}
							>
								{t('modals.save-theme.desc')}
							</TextNative>
						</View>
						<View
							style={[
								border.radius.medium,
								padding.left.small,
								margin.top.medium,
								{ backgroundColor: colors['input-background'] },
							]}
						>
							<TextInput
								value={themeName}
								multiline
								onChange={({ nativeEvent }) => setThemeName(nativeEvent.text)}
								style={[
									text.bold.small,
									{
										fontFamily: 'Open Sans',
										color: colors['background-header'],
										paddingRight: 12 * scaleSize,
									},
								]}
								placeholder={t('modals.save-theme.placeholder')}
								placeholderTextColor={colors['secondary-text']}
							/>
						</View>
					</View>
				</View>
				<View style={[row.center, margin.top.small]}>
					<TouchableOpacity
						style={[
							margin.bottom.medium,
							opacity(0.5),
							_styles.skipButton,
							{ flexDirection: 'row', justifyContent: 'center', alignItems: 'center' },
						]}
						onPress={async () => {
							closeModal()
						}}
					>
						<Icon
							name='close'
							width={30}
							height={30}
							fill={colors['negative-asset']}
							style={row.item.justify}
						/>
						<TextNative
							style={[
								padding.left.small,
								row.item.justify,
								text.size.scale(16),
								text.bold.medium,
								{
									fontFamily: 'Open Sans',
									color: colors['negative-asset'],
									textTransform: 'uppercase',
								},
							]}
						>
							{t('generic.cancel')}
						</TextNative>
					</TouchableOpacity>
					<TouchableOpacity
						style={[
							margin.bottom.medium,
							_styles.addButton,
							{
								flexDirection: 'row',
								justifyContent: 'center',
								alignItems: 'center',
								backgroundColor: colors['positive-asset'],
							},
						]}
						onPress={async () => {
							await ctx.setPersistentOption({
								type: PersistentOptionsKeys.ThemeColor,
								payload: {
									selected: themeName,
									collection: {
										...ctx.persistentOptions.themeColor.collection,
										[themeName]: {
											colors:
												ctx.persistentOptions.themeColor.collection[CurrentGeneratedTheme].colors,
										},
									},
									isDark: themeName === DefaultDarkTheme,
								},
							})
							closeModal()
						}}
					>
						<Icon
							name='checkmark-outline'
							width={30}
							height={30}
							fill={colors['background-header']}
							style={row.item.justify}
						/>
						<TextNative
							style={[
								padding.left.small,
								row.item.justify,
								text.size.scale(16),
								text.bold.medium,
								{ color: colors['background-header'] },
							]}
						>
							{t('modals.save-theme.add')}
						</TextNative>
					</TouchableOpacity>
				</View>
			</View>
		</View>
	)
}

export const ThemeColorName: React.FC<{ closeModal: () => void }> = ({ closeModal }) => {
	const [{}, { windowHeight }] = useStyles()

	return (
		<View style={[StyleSheet.absoluteFill, { elevation: 6, zIndex: 6 }]}>
			<TouchableOpacity
				activeOpacity={0}
				style={[
					StyleSheet.absoluteFill,
					{
						position: 'absolute',
						top: 0,
						bottom: 0,
						left: 0,
						right: 0,
						height: windowHeight,
					},
				]}
				onPress={async () => {
					closeModal()
				}}
			>
				<BlurView style={[StyleSheet.absoluteFill]} blurType='light' />
			</TouchableOpacity>
			<ThemeColorBody closeModal={closeModal} />
		</View>
	)
}

export default ThemeColorName
