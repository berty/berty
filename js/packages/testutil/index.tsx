import React from 'react'
import { NavigationContainer } from '@react-navigation/native'
import { IconRegistry } from '@ui-kitten/components'
import { EvaIconsPack } from '@ui-kitten/eva-icons'
import { SafeAreaProvider } from 'react-native-safe-area-context'

import { Provider as ThemeProvider } from '@berty-tech/components/theme'
import { FeatherIconsPack } from '@berty-tech/messenger-app/feather-icons'
import { CustomIconsPack } from '@berty-tech/messenger-app/custom-icons'
import { Provider as StyleProvider } from '@berty-tech/styles'

export const TestProvider: React.FC = ({ children }) => (
	<SafeAreaProvider
		initialMetrics={{
			frame: { x: 0, y: 0, width: 0, height: 0 },
			insets: { top: 0, left: 0, right: 0, bottom: 0 },
		}}
	>
		<StyleProvider>
			<IconRegistry icons={[EvaIconsPack, FeatherIconsPack, CustomIconsPack]} />
			<ThemeProvider>
				<NavigationContainer>{children}</NavigationContainer>
			</ThemeProvider>
		</StyleProvider>
	</SafeAreaProvider>
)

import _fixtures from './fixtures'
export const fixtures = _fixtures
