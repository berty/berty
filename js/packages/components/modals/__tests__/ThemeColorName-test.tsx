import React from 'react'
import renderer from 'react-test-renderer'

import { TestProvider } from '@berty-tech/testutil'

import ThemeColorName from '../ThemeColorName'

test('ThemeColorName modal renders correctly', () => {
	const props: React.ComponentProps<typeof ThemeColorName> = {
		closeModal: () => {},
	}
	const tree = renderer
		.create(
			<TestProvider>
				<ThemeColorName {...props} />
			</TestProvider>,
		)
		.toJSON()
	expect(tree).toMatchSnapshot()
})
