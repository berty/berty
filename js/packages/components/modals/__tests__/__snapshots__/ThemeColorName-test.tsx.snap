// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeColorName modal renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "elevation": 6,
            "zIndex": 6,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "bottom": 0,
            "height": 1334,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <BlurView
          blurAmount={10}
          blurType="light"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ],
            ]
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 250,
              "justifyContent": "center",
              "top": "25%",
            },
            Object {
              "margin": 32,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "elevation": 7,
                "height": 110,
                "justifyContent": "center",
                "position": "relative",
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 3,
                  "width": 0,
                },
                "shadowOpacity": 0.1,
                "shadowRadius": 5,
                "top": 50,
                "width": 110,
                "zIndex": 1,
              },
              Object {
                "borderRadius": 60,
              },
            ]
          }
        >
          <SvgMock
            height={125}
            width={125}
          />
        </View>
        <View
          style={
            Array [
              Object {
                "paddingHorizontal": 16,
              },
              Object {
                "paddingBottom": 16,
              },
              Object {
                "borderRadius": 24,
              },
              Object {
                "shadowOffset": Object {
                  "height": 6.5,
                  "width": 0,
                },
                "shadowOpacity": 0.2,
                "shadowRadius": 13,
              },
              Object {
                "backgroundColor": "#FFFFFF",
                "shadowColor": "#000000",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "marginTop": 70,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "textAlign": "center",
                  },
                  Object {
                    "paddingTop": 9,
                  },
                  Object {
                    "fontSize": 38,
                  },
                  Object {
                    "fontWeight": "bold",
                  },
                  Object {
                    "color": "#383B62",
                    "fontFamily": "Open Sans",
                  },
                ]
              }
            >
              ðŸŽ¨ modals.save-theme.title
            </Text>
            <View
              style={
                Array [
                  Object {
                    "paddingTop": 20,
                  },
                  Object {
                    "paddingHorizontal": 16,
                  },
                  Object {
                    "flexDirection": "column",
                  },
                ]
              }
            >
              <View>
                <Text
                  style={
                    Array [
                      Object {
                        "fontWeight": "500",
                      },
                      Object {
                        "fontSize": 30,
                      },
                      Object {
                        "color": "#383B62",
                        "fontFamily": "Open Sans",
                      },
                    ]
                  }
                >
                  modals.save-theme.desc
                </Text>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "borderRadius": 16,
                    },
                    Object {
                      "paddingLeft": 9,
                    },
                    Object {
                      "marginTop": 16,
                    },
                    Object {
                      "backgroundColor": "#F7F8FF",
                    },
                  ]
                }
              >
                <TextInput
                  allowFontScaling={true}
                  multiline={true}
                  onChange={[Function]}
                  placeholder="modals.save-theme.placeholder"
                  placeholderTextColor="#979797"
                  rejectResponderTermination={true}
                  style={
                    Array [
                      Object {
                        "fontWeight": "500",
                      },
                      Object {
                        "color": "#525BEC",
                        "fontFamily": "Open Sans",
                        "paddingRight": 12,
                      },
                    ]
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "stretch",
                  "flexDirection": "row",
                  "justifyContent": "space-evenly",
                },
                Object {
                  "marginTop": 9,
                },
              ]
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#A5A7C5",
                  "borderRadius": 9,
                  "borderWidth": 2,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginBottom": 16,
                  "marginTop": 15,
                  "opacity": 0.5,
                  "paddingBottom": 9,
                  "paddingLeft": 9,
                  "paddingRight": 16,
                  "paddingTop": 9,
                  "width": 120,
                }
              }
            >
              <View
                style={
                  Object {
                    "transform": Array [
                      Object {
                        "scale": 1,
                      },
                    ],
                  }
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight={30}
                  bbWidth={30}
                  fill="#A5A7C5"
                  focusable={false}
                  height={30}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Object {
                        "alignSelf": "center",
                      },
                      Object {
                        "flex": 0,
                        "height": 30,
                        "width": 30,
                      },
                    ]
                  }
                  vbHeight={24}
                  vbWidth={24}
                  width={30}
                >
                  <RNSVGGroup
                    fill={4289046469}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGGroup>
                      <RNSVGGroup>
                        <RNSVGRect
                          height="24"
                          matrix={
                            Array [
                              -1,
                              1.2246467991473532e-16,
                              -1.2246467991473532e-16,
                              -1,
                              24,
                              24,
                            ]
                          }
                          opacity={0}
                          width="24"
                          x={0}
                          y={0}
                        />
                        <RNSVGPath
                          d="M13.41 12l4.3-4.29a1 1 0 1 0-1.42-1.42L12 10.59l-4.29-4.3a1 1 0 0 0-1.42 1.42l4.3 4.29-4.3 4.29a1 1 0 0 0 0 1.42 1 1 0 0 0 1.42 0l4.29-4.3 4.29 4.3a1 1 0 0 0 1.42 0 1 1 0 0 0 0-1.42z"
                        />
                      </RNSVGGroup>
                    </RNSVGGroup>
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
              <Text
                style={
                  Array [
                    Object {
                      "paddingLeft": 9,
                    },
                    Object {
                      "alignSelf": "center",
                    },
                    Object {
                      "fontSize": 32,
                    },
                    Object {
                      "fontWeight": "bold",
                    },
                    Object {
                      "color": "#A5A7C5",
                      "fontFamily": "Open Sans",
                    },
                  ]
                }
              >
                modals.save-theme.cancel
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#CED2FF",
                  "borderColor": "#CED2FF",
                  "borderRadius": 9,
                  "borderWidth": 2,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginBottom": 16,
                  "marginTop": 15,
                  "opacity": 1,
                  "paddingBottom": 9,
                  "paddingLeft": 9,
                  "paddingRight": 16,
                  "paddingTop": 9,
                  "width": 120,
                }
              }
            >
              <View
                style={
                  Object {
                    "transform": Array [
                      Object {
                        "scale": 1,
                      },
                    ],
                  }
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight={30}
                  bbWidth={30}
                  fill="#525BEC"
                  focusable={false}
                  height={30}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Object {
                        "alignSelf": "center",
                      },
                      Object {
                        "flex": 0,
                        "height": 30,
                        "width": 30,
                      },
                    ]
                  }
                  vbHeight={24}
                  vbWidth={24}
                  width={30}
                >
                  <RNSVGGroup
                    fill={4283587564}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGGroup>
                      <RNSVGGroup>
                        <RNSVGRect
                          height="24"
                          opacity={0}
                          width="24"
                          x={0}
                          y={0}
                        />
                        <RNSVGPath
                          d="M9.86 18a1 1 0 0 1-.73-.32l-4.86-5.17a1 1 0 1 1 1.46-1.37l4.12 4.39 8.41-9.2a1 1 0 1 1 1.48 1.34l-9.14 10a1 1 0 0 1-.73.33z"
                        />
                      </RNSVGGroup>
                    </RNSVGGroup>
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
              <Text
                style={
                  Array [
                    Object {
                      "paddingLeft": 9,
                    },
                    Object {
                      "alignSelf": "center",
                    },
                    Object {
                      "fontSize": 32,
                    },
                    Object {
                      "fontWeight": "bold",
                    },
                    Object {
                      "color": "#525BEC",
                    },
                  ]
                }
              >
                modals.save-theme.add
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
